### PR TITLE
Port v0.12.3 fixes

### PR DESCRIPTION
**Pull request by lorenzo-dev1**
Ports v0.12.3 temporary fixes on:
- prometheus incompatibility with buffered gauges
- message cache size

### CODE REVIEW

**Review by yazzaoui — APPROVED** [review]
Thanks for that 👍

**Review by Jason-Zhangxin-Chen — APPROVED** [review]
LGTM

File: consensus/tendermint/backend/engine.go [hunk 1]
```diff
@@ -29,8 +29,8 @@ import (
 
 const (
 	inmemorySnapshots = 128 // Number of recent vote snapshots to keep in memory
-	inmemoryPeers     = 40
-	inmemoryMessages  = 1024
+	inmemoryPeers     = 150
+	inmemoryMessages  = 8192
```
Comment (vpiyush) [inline consensus/tendermint/backend/engine.go:33]: For my curiousity, Why do we need this so high? this is going to increase live heap size.

Comment (lorenzo-dev1) [reply consensus/tendermint/backend/engine.go:33]: I leave this one for @yazzaoui but I think it is just trial and error to see how big we need to make it to not have the repeated messages issue.

Comment (yazzaoui) [reply consensus/tendermint/backend/engine.go:33]: TBH even we those values we still have duplicate message warnings. I don't think we have time to solve the caching for the next release but we should think about a new approach after. For now let's leave it like that.

**Pull request merged**